tempesta_fw/http.c

```diff
@@ -660,24 +673,29 @@ tfw_http_req_zap_error(struct list_head *equeue)
 
 	list_for_each_entry_safe(req, tmp, equeue, fwd_list) {
 		list_del_init(&req->fwd_list);
-		switch(req->httperr.status) {
-		case 404:
-			tfw_http_send_404(req, req->httperr.reason);
-			break;
-		case 500:
-			tfw_http_send_500(req, req->httperr.reason);
-			break;
-		case 502:
-			tfw_http_send_502(req, req->httperr.reason);
-			break;
-		case 504:
-			tfw_http_send_504(req, req->httperr.reason);
-			break;
-		default:
-			TFW_WARN("Unexpected response error code: [%d]\n",
-				 req->httperr.status);
-			tfw_http_send_500(req, req->httperr.reason);
-			break;
+		if (!ret) {
+			switch(req->httperr.status) {
+			case 404:
+				ret = tfw_http_send_404(req, req->httperr.reason);
+				break;
+			case 500:
+				ret = tfw_http_send_500(req, req->httperr.reason);
+				break;
+			case 502:
+				ret = tfw_http_send_502(req, req->httperr.reason);
+				break;
+			case 504:
+				ret = tfw_http_send_504(req, req->httperr.reason);
```

Comment:
The lines are bit wider than 80 characters and there is no need to place `switch` inside the `if` statement: just more statistic accounting before sending the responses and use `continue` statement in if.

Reply:
Corrected: error handling logic has been removed from the tfw_http_req_zap_error function.